include/system/analytics_manager.js

```diff
@@ -181,30 +399,48 @@ module.exports = function AnalyticsManagerModule(pb) {
      * providers.
      */
     AnalyticsManager.onPageRender = function(req, session, ls, cb) {
-        var site = pb.RequestHandler.sites[req.headers.host] ? pb.RequestHandler.sites[req.headers.host].uid : null;
-        if (checkForHook(site)) {
-            //short circuit when there are not tasks for the site.
-            return cb(null, '');
-        }
-        var managerInstance = new AnalyticsManager({ site: site });
+        var context = {
+            site: pb.RequestHandler.sites[req.headers.host] ? pb.RequestHandler.sites[req.headers.host].uid : null,
+            timeout: pb.config.analytics.timeout
+        };
+        var managerInstance = new AnalyticsManager(context);
         managerInstance.gatherData(req, session, ls, cb);
     };
 
-    function checkForHook(site) {
-        return !PROVIDER_HOOKS[pb.SiteService.GLOBAL_SITE] ||
-            Object.keys(PROVIDER_HOOKS[pb.SiteService.GLOBAL_SITE]).length === 0
-        !PROVIDER_HOOKS[site] ||
-        Object.keys(PROVIDER_HOOKS[site]).length === 0;
-    }
+    /**
+     * @static
+     * @method getKeys
+     * @param {String} site
+     * @return {Array}
+     */
+    AnalyticsManager.getKeys = function(site) {
+        var keyHash = {};
+        var keyFunc = keyHandler(keyHash);
+
+        //check for global keys.  They apply to all sites
+        if (PROVIDER_HOOKS[pb.SiteService.GLOBAL_SITE]) {
+            (Object.keys(PROVIDER_HOOKS[pb.SiteService.GLOBAL_SITE]) || []).forEach(keyFunc);
+        }
 
-    function getKeys(site) {
-        var keyArray = Object.keys(PROVIDER_HOOKS[pb.SiteService.GLOBAL_SITE]) || [];
-        if(PROVIDER_HOOKS[site] && Object.keys(PROVIDER_HOOKS[site]).length > 0) {
-            return keyArray.concat(Object.keys(PROVIDER_HOOKS[site]))
+        //check for non-global site keys
+        if(PROVIDER_HOOKS[site] && site !== pb.SiteService.GLOBAL_SITE) {
+            Object.keys(PROVIDER_HOOKS[site]).forEach(keyFunc);
         }
-        return keyArray;
+        return Object.keys(keyHash);
     }
```

Comment:
There was a small bug when in single site mode.  It would call a provider twice because it did not verify that the provider had already been added if it belonged to the global site too.  A hash is built of the provider keys from both then the distinct union of the two is used as the array of keys that is returned.